go-controller/pkg/ovn/subnetallocator/allocator.go

```diff
@@ -108,21 +160,36 @@ func (sna *SubnetAllocator) AllocateIPv6Network() (*net.IPNet, error) {
 	return nil, ErrSubnetAllocatorFull
 }
 
-func (sna *SubnetAllocator) ReleaseNetwork(subnet *net.IPNet) error {
+func (sna *BaseSubnetAllocator) ReleaseNetworks(subnets []*net.IPNet) error {
 	sna.Lock()
 	defer sna.Unlock()
+	return sna.releaseNetworks(subnets)
+}
 
-	for _, snr := range sna.v4ranges {
-		if snr.releaseNetwork(subnet) {
-			return nil
+func (sna *BaseSubnetAllocator) releaseNetworks(subnets []*net.IPNet) error {
+	var errorList []error
+	for _, subnet := range subnets {
+		released := false
+		for _, snr := range sna.v4ranges {
+			if snr.releaseNetwork(subnet) {
+				released = true
+				break
+			}
 		}
-	}
-	for _, snr := range sna.v6ranges {
-		if snr.releaseNetwork(subnet) {
-			return nil
+		if !released {
+			for _, snr := range sna.v6ranges {
+				if snr.releaseNetwork(subnet) {
+					released = true
+					break
+				}
+			}
+		}
+		if !released {
+			errorList = append(errorList, fmt.Errorf("network %s does not belong to any known range", subnet.String()))
 		}
 	}
-	return fmt.Errorf("network %s does not belong to any known range", subnet.String())
```

Comment:
I believe I wanted to make sure the error still got surfaced. It would be an odd situation, like maybe if somebody modified node annotations while the node was up. But the error itself is kinda pointless.

I made the error itself, once passed up to deleteNode() in master.go a log-and-continue error, since it doesn't make sense to block node cleanup on this as you point out.